src/RamTransforms.cpp

```diff
@@ -255,24 +257,134 @@ std::unique_ptr<RamExpression> MakeIndexTransformer::getExpression(
                 const RamExpression* rhs = &binRelOp->getRHS();
                 if (lhs->getTupleId() == identifier && rla->getLevel(rhs) < identifier) {
                     element = lhs->getElement();
-                    return std::unique_ptr<RamExpression>(rhs->clone());
+                    return std::make_pair(std::unique_ptr<RamExpression>(rhs->clone()),
```

Comment:
Please use `souffle::clone` instead of `std::unique_ptr<.....>(blah->clone())`. If you're not in an AST member this can be further simplified as just `clone`. e.g. `return std::make_pair(std::unique_ptr<RamExpression>(rhs->clone()), std::unique_ptr<RamExpression>(lhs->clone()))` -> `return {clone(rhs), clone(lhs)};`

Reply:
And is this intended to swap the LHS/RHS? If so, could you add a comment noting why?

Reply:
`std::make_pair` seems a bit overkill when you can do `{ a, b }` in type deducible contexts.